### PR TITLE
Views V2 shortcode support

### PR DESCRIPTION
**Pull request by bordoni**
_Ref:_ [C#127995](http://central.tri.be/issues/127995)
---

Depends #2556 https://github.com/moderntribe/events-pro/pull/1161

### CODE REVIEW

File: src/Tribe/Views/V2/View.php [hunk 1]
```diff
@@ -201,10 +203,13 @@ public static function make_for_rest( \WP_REST_Request $request ) {
 				)
 			);
 
-		$view =  static::make( $slug, $context );
+		$view = static::make( $slug, $context );
 
 		$view->url = $url_object;
 
+		// Setup wether this view should manage url or not, based on the Rest Request Sent.
```
Comment (borkweb) [inline src/Tribe/Views/V2/View.php]:
```suggestion
		// Setup whether this view should manage URL or not, based on the Rest Request Sent.
```

File: src/resources/js/views/manager.js [hunk 2]
```diff
@@ -72,8 +72,15 @@ tribe.events.views.manager = {};
 	obj.setup = function( index, container ) {
 		var $container = $( container );
 		var $form = $container.find( obj.selectors.form );
+		var $data = $container.find( obj.selectors.dataScript );
+		var data = {};
```
Comment (borkweb) [inline src/resources/js/views/manager.js]:
```suggestion
		var data  = {};
```

File: src/resources/js/views/manager.js [hunk 3]
```diff
@@ -107,6 +114,30 @@ tribe.events.views.manager = {};
...
+	 *
+	 * @since TBD
+	 *
+	 * @param  {Element|jQuery} element Which element we are using as the container.
+	 *
+	 * @return {Boolean}
+	 */
+	obj.shouldManageUrl = function( $container ) {
+		var shouldManageUrl = $container.data( 'view-manage-url' );
+		var tribeIsTruthy = /^(true|1|on|yes)$/;
```
Comment (borkweb) [inline src/resources/js/views/manager.js]:
```suggestion
		var tribeIsTruthy   = /^(true|1|on|yes)$/;
```

File: src/views/v2/events-bar/form.php [hunk 4]
```diff
@@ -12,13 +12,18 @@
...
+
+use Tribe\Events\Views\V2\Rest_Endpoint;
 ?>
 <div class="tribe-events-c-events-bar__form">
 	<form
 		class="tribe-common-c-search"
 		method="get"
 		data-js="tribe-events-view-form"
 	>
+		<?php wp_nonce_field( 'wp_rest', 'tribe-events-views[_wpnonce]' ); ?>
```
Comment (borkweb) [inline src/views/v2/events-bar/form.php]: I'd suggest the action be something other than `wp_rest` as the view won't always hit a REST endpoint.

File: src/views/v2/top-bar/today.php [hunk 5]
```diff
@@ -12,9 +12,20 @@
  * @version 4.9.3
  *
  */
+
+// If we didnt have a view setup we cannot print what today's link
```
Comment (borkweb) [inline src/views/v2/top-bar/today.php]:
```suggestion
// If we didn't have a view setup we cannot print today's link
```

File: src/views/v2/events-bar/form.php (hunk 4 above)
Comment (bordoni) [reply src/views/v2/events-bar/form.php]: Agreed, that is something we will need to address by having this be a method inside of the of another class later, or template.

**Pull request merged**